probes/sastToolRunsOnAllCommits/impl_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package sastToolRunsOnAllCommits
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+func Test_Run(t *testing.T) {
+	t.Parallel()
+	// nolint:govet
+	tests := []struct {
+		name             string
+		raw              *checker.RawResults
+		outcomes         []finding.Outcome
+		err              error
+		expectedFindings []finding.Finding
+	}{
+		{
+			name: "sonar present",
+			err:  nil,
+			raw: &checker.RawResults{
+				SASTResults: checker.SASTData{
+					Commits: []checker.SASTCommit{
+						{
+							Compliant: false,
+						},
+						{
+							Compliant: true,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
+			},
+			expectedFindings: []finding.Finding{
+				{
+					Probe:   "sastToolRunsOnAllCommits",
+					Message: "1 commits out of 2 are checked with a SAST tool",
+					Values: map[string]int{
+						"totalTested": 1,
```

Comment:
can we be more explicit about what is tested and merged? Something like `totalPullRequestsAnalyzed` and `totlalPullRequestsMerged`?

Reply:
Done in https://github.com/ossf/scorecard/pull/3571/commits/27d970def7802bd22f9824a6424cbfebfaf12a85